### PR TITLE
Fix capability granting for auto-properties

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1587475

### CODE REVIEW

**Comment by tmat** [comment]
@davidwengier PTAL

File: src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs [hunk 1]
```diff
@@ -4281,23 +4288,21 @@ static bool IsSecurityAttribute(INamedTypeSymbol namedTypeSymbol)
...
             }
-            else if (newSymbol is IFieldSymbol field)
-            {
-                if (field.IsStatic)
-                {
-                    return capabilities.Grant(EditAndContinueCapabilities.AddStaticFieldToExistingType);
-                }
 
-                return capabilities.Grant(EditAndContinueCapabilities.AddInstanceFieldToExistingType);
+            if (newSymbol is IFieldSymbol || newSymbol is IPropertySymbol { DeclaringSyntaxReferences: [var syntaxRef] } && HasBackingField(syntaxRef.GetSyntax(cancellationToken)))
```
Comment (davidwengier) [inline src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs:4300]: Nice list pattern!

File: src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs [hunk 2]
```diff
@@ -14361,6 +14361,19 @@ public void PropertyInsert()
                 capabilities: EditAndContinueCapabilities.AddMethodToExistingType);
         }
 
+        [Fact]
+        public void Property_Insert_Static()
```
Comment (davidwengier) [inline src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs:14365]: Perhaps add a test for a static auto-prop, just for completeness?

**Review by davidwengier — APPROVED** [review]

File: src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs (hunk 2 above)
Comment (tmat) [reply src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs:14365]: There is one: Property_Insert_Auto_Static

Comment (davidwengier) [reply src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs:14365]: Whoops, right you are. Not sure how I missed that earlier :)

**Pull request merged**